matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt

```diff
@@ -41,10 +61,54 @@ internal class RoomMemberEventHandler @Inject constructor() {
         }
         val roomMemberEntity = RoomMemberEntityFactory.create(roomId, userId, roomMember)
         realm.insertOrUpdate(roomMemberEntity)
-        if (roomMember.membership.isActive()) {
-            val userEntity = UserEntityFactory.create(userId, roomMember)
-            realm.insertOrUpdate(userEntity)
+
+        if (roomMember.membership.isActive() && shouldUpdateUserEntity(realm, userId, roomMember)) {
+            updateUserEntity(realm, userId, roomMember)
+            if (userId == this.userId) {
+                fetchUserProfile(userId) // To fix #1715 (myroomnick changes the global name)
+            }
         }
         return true
     }
+
+    private fun shouldUpdateUserEntity(realm: Realm, userId: String, roomMember: RoomMemberContent?): Boolean {
```

Comment:
You are fetching an entity to check for changes and then we use insertOrUpdate, so it's fetching again. 
Please re-use the UserEntity directly to update its fields

Reply:
Fixed by reusing the entity, I want to discuss using RealmLiveEntityObserver.